pkg/kv/kvserver/spanlatch/manager.go

```diff
@@ -503,22 +510,64 @@ func (m *Manager) iterAndWait(
 		if ignore(wait.ts, held.ts) {
 			continue
 		}
-		if err := m.waitForSignal(ctx, t, waitType, heldType, wait, held); err != nil {
+		if err := m.waitForSignal(ctx, t, pp, waitType, heldType, wait, held); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// A PoisonPolicy determines how a request will react to encountering a poisoned
+// latch.
+type PoisonPolicy byte
```

Comment:
Can this be moved to pkg/kv/kvserver/concurrency? Or does that pull in spanlatch?